src/mbgl/style/layers/circle_layer.cpp

```diff
@@ -393,5 +400,344 @@ TransitionOptions CircleLayer::getCircleStrokeOpacityTransition() const {
     return impl().paint.template get<CircleStrokeOpacity>().options;
 }
 
+using namespace conversion;
+
+optional<Error> CircleLayer::setPaintProperty(const std::string& name, const Convertible& value) {
+    enum class Property {
+        Unknown,
+        CircleRadius,
+        CircleColor,
+        CircleBlur,
+        CircleOpacity,
+        CircleTranslate,
+        CircleTranslateAnchor,
+        CirclePitchScale,
+        CirclePitchAlignment,
+        CircleStrokeWidth,
+        CircleStrokeColor,
+        CircleStrokeOpacity,
+        CircleRadiusTransition,
+        CircleColorTransition,
+        CircleBlurTransition,
+        CircleOpacityTransition,
+        CircleTranslateTransition,
+        CircleTranslateAnchorTransition,
+        CirclePitchScaleTransition,
+        CirclePitchAlignmentTransition,
+        CircleStrokeWidthTransition,
+        CircleStrokeColorTransition,
+        CircleStrokeOpacityTransition,
+    };
+
+    Property property = Property::Unknown;
+    switch (util::hashFNV1a(name.c_str())) {
+    case util::hashFNV1a("circle-radius"):
+        if (name == "circle-radius") {
```

Comment:
Is this additional string comparison needed because `util::hashFNV1a("circle-radius")` could result in a false positive for e.g. `"circle-radius-transition"`?

Reply:
Yes, correct. It's probably a perfect hash for the set of property names, but the `name` input is an arbitrary string.